src/main/java/com/hazelcast/aws/AwsDiscoveryStrategy.java

```diff
@@ -132,10 +136,40 @@ private String getPortRange() {
             String availabilityZone = awsClient.getAvailabilityZone();
             LOGGER.info(String.format("Availability zone found: '%s'", availabilityZone));
             memberMetadata.put(PartitionGroupMetaData.PARTITION_GROUP_ZONE, availabilityZone);
+
+            getPlacementGroup().ifPresent(pg ->
+                    memberMetadata.put(PARTITION_GROUP_PLACEMENT, availabilityZone + '-' + pg));
         }
         return memberMetadata;
     }
 
+    /**
+     * Resolves the placement group of the resource if it belongs to any.
+     * <p>
+     * If the placement group is Cluster Placement Group or Spread Placement Group, then returns
+     * the group name. If it is Partition Placement Group, then returns the group name with the
+     * partition number prefixed by {@code '@'} appended.
+     * <p>
+     * When forming partition groups, this name should be combined with zone name. Otherwise
+     * two resources in different zones but in the same placement group will be assumed as
+     * a single group.
+     *
+     * @see AwsClient#getPlacementGroup()
+     * @see AwsClient#getPlacementPartitionNumber()
+     * @return  Placement group name if exists, empty otherwise.
+     */
+    private Optional<String> getPlacementGroup() {
+        Optional<String> placementGroup = awsClient.getPlacementGroup();
+        if (!placementGroup.isPresent()) {
+            LOGGER.fine("No placement group is found.");
+            return Optional.empty();
+        }
+        StringBuilder result = new StringBuilder(placementGroup.get());
+        awsClient.getPlacementPartitionNumber().ifPresent(ppn -> result.append('@').append(ppn));
+        LOGGER.info(String.format("Placement group found: '%s'", result.toString()));
```

Comment:
So, I guess this is printed only when you set Placement Group in AWS. Otherwise, nothing is printed at the "info" level. Right?

Reply:
I thought if we log at info level saying: "no placement group found", it could be misleading such that instances **should** be a part of a placement group. That was the reason I kept it at the fine level. However, we can consider logging it at info. WDYT?

Reply:
I think the behevior should be like this:
- No placement groups used => Log at fine level
- Placement group used and successful group name retrieval => Log at info
- Placement group used and no successful group name retrieval => warning or fast failing

I guess it works that way. My comment was just to confirm it :)

Reply:
Oops, sorry. I assumed you were going to request some changes with the following comment. 

The behavior is true for the 1st and the 2nd. But for the 3rd one it's a bit different. We check if a placement group is used or not via a metadata endpoint that exists only for the VMs having a placement group. If the response code is 200 then we assume a placement group exists and use the received data. Otherwise, we do not check if the response code is 404 or not - and the discovery starts without placement metadata.

So it's a bit tricky - yet possible, to realize that "Placement group is used but no successful group name retrieval occurred". I'm not sure if we should even consider this case as I feel it must be ensured by AWS.

Reply:
I see. But if you specify `PLACEMENT_GROUP_AWARE` in the Hazelcast configuration, but the retrieval of placement group in Hazelcast AWS plugin fails, then Hazelcast fails fast to start. That's how I understand [this code](https://github.com/hazelcast/hazelcast/pull/18048/files#diff-b0f2cd2392e82d724f7c814ee9797a082c5df5b9ead0761eadea41706d935d23R48). Am I correct?

Reply:
Exactly